From 1134921f50b6383a77eaf87fae8f3de1f90467d5 Mon Sep 17 00:00:00 2001
From: Richard Liu <r66033@freescale.com>
Date: Wed, 4 Dec 2013 16:49:53 +0800
Subject: [PATCH 480/498] ENGR00290626 enable hwcomposer 1.3 on KitKat

define HWC API version to 1.3 on KitKat,
it will benefit virtual display performance.

Signed-off-by: Richard Liu <r66033@freescale.com>
---
 mx6/hwcomposer/Android.mk     | 3 ---
 mx6/hwcomposer/hwcomposer.cpp | 6 +-----
 2 files changed, 1 insertion(+), 8 deletions(-)

diff --git a/mx6/hwcomposer/Android.mk b/mx6/hwcomposer/Android.mk
index af9aee1..6cee365 100755
--- a/mx6/hwcomposer/Android.mk
+++ b/mx6/hwcomposer/Android.mk
@@ -41,9 +41,6 @@ LOCAL_MODULE := hwcomposer.$(TARGET_BOARD_PLATFORM)
 LOCAL_C_INCLUDES += hardware/imx/mx6/libgralloc_wrapper
 LOCAL_CFLAGS:= -DLOG_TAG=\"hwcomposer\"
 LOCAL_CFLAGS += -DENABLE_VSYNC
-ifneq ($(HAVE_FSL_IMX_GPU3D),true)
-LOCAL_CFLAGS += -DUSE_HWCOMPOSER_VERSION_1_2
-endif
 LOCAL_MODULE_TAGS := optional
 include $(BUILD_SHARED_LIBRARY)
 endif
diff --git a/mx6/hwcomposer/hwcomposer.cpp b/mx6/hwcomposer/hwcomposer.cpp
index 5fd45f0..03cdfd7 100755
--- a/mx6/hwcomposer/hwcomposer.cpp
+++ b/mx6/hwcomposer/hwcomposer.cpp
@@ -324,11 +324,7 @@ static int hwc_device_open(const struct hw_module_t* module, const char* name,
 
         dev->device.prepare = hwc_prepare;
         dev->device.set = hwc_set;
-#ifdef  USE_HWCOMPOSER_VERSION_1_2
-        dev->device.common.version = HWC_DEVICE_API_VERSION_1_2;
-#else
-        dev->device.common.version = HWC_DEVICE_API_VERSION_1_1;
-#endif
+        dev->device.common.version = HWC_DEVICE_API_VERSION_1_3;
         dev->device.registerProcs = hwc_registerProcs;
         dev->device.eventControl = hwc_eventControl;
         dev->device.query = hwc_query;
-- 
1.8.0

